test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/DerivedTypes.cs

```diff
@@ -0,0 +1,360 @@
+        {
+            throw (Exception)Activator.CreateInstance(ExceptionType, Message);
+        }
+
+        AuthenticationProperties ISecureDataFormat<AuthenticationProperties>.Unprotect(string protectedText)
+        {
+            throw (Exception)Activator.CreateInstance(ExceptionType, Message);
+        }
+    }
+
+    /// <summary>
+    /// This formatter returns values passed in the constructor
+    /// Defaults:
+    /// Protect: null
+    /// UnProtect: null
+    /// </summary>
+    public class AuthenticationPropertiesFormaterSetReturn : ISecureDataFormat<AuthenticationProperties>
+    {
+        string _protect;
+        AuthenticationProperties _unprotect;
+
+        AuthenticationPropertiesFormaterSetReturn(string protect = null, AuthenticationProperties unprotect = null)
+        {
+            _protect = protect;
+            _unprotect = unprotect;
+        }
+
+        public string Protect(AuthenticationProperties data)
+        {
+            return _protect;
+        }
+
+        AuthenticationProperties ISecureDataFormat<AuthenticationProperties>.Unprotect(string protectedText)
+        {
+            return _unprotect;
+        }
+    }
+
+    public class CustomConfigureOptions : ConfigureOptions<OpenIdConnectAuthenticationOptions>
+    {
+        public CustomConfigureOptions(Action<OpenIdConnectAuthenticationOptions> action)
+            : base(action)
+        {
+        }
+
+        public override void Configure(OpenIdConnectAuthenticationOptions options, string name = "")
+        {
+            base.Configure(options, name);
+            return;
+        }
+    }
+
+    public class CustomLogger : ILogger, IDisposable
+    {
+        LogLevel _logLevel = 0;
+
+        public CustomLogger(LogLevel logLevel = LogLevel.Debug)
+        {
+            _logLevel = logLevel;
+        }
+
+        List<LogEntry> _logEntries = new List<LogEntry>();
+
+        public IDisposable BeginScopeImpl(object state)
+        {
+            return this;
+        }
+
+        public void Dispose()
+        {
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return (logLevel >= _logLevel);
+        }
+
+        public void Log(LogLevel logLevel, int eventId, object state, Exception exception, Func<object, Exception, string> formatter)
+        {
+            if (IsEnabled(logLevel))
+            {                
+                var logEntry =
+                    new LogEntry
+                    {
+                        EventId = eventId,
+                        Exception = exception,
+                        Formatter = formatter,
+                        Level = logLevel,
+                        State = state,
+                    };
+
+                _logEntries.Add(logEntry);
+                Debug.WriteLine(logEntry.ToString());
+            }
+        }
+
+        public List<LogEntry> Logs { get { return _logEntries; } }
+    }
+
+    public class CustomLoggerFactory : ILoggerFactory
```

Comment:
I am not actually testing a custom logger, but am using a custom logger to obtain and compare logs to ensure correctness.